src/make.ts

```diff
@@ -95,15 +95,58 @@ function error<T>(error: string): Make<T> {
   return Make.error<T>([{ path: [], error }]);
 }
 
-function checkString(value: any) {
-  if (typeof value !== 'string') {
-    return getErrorWithValueMsg('expected a string', value);
+export function registerFormat(format: string, maker: Maker<any, undefined>) {
```

Comment:
So is this supposed to be called by whatever is implementing oats, to use "fancier" formatters?

Reply:
yes. openapi3 spec leaves the format field pretty much free for all https://swagger.io/docs/specification/data-models/data-types/#string

Reply:
Should we add support to these

> date – full-date notation as defined by RFC 3339, section 5.6, for example, 2017-07-21
date-time – the date-time notation as defined by RFC 3339, section 5.6, for example, 2017-07-21T17:32:28Z
password – a hint to UIs to mask the input
byte – base64-encoded characters, for example, U3dhZ2dlciByb2Nrcw==
binary – binary data, used to describe files (see Files below)

built-in? I could make a PR at some point

Reply:
binary we kinda support. but not others. I guess dates would be most usable.